runtime/src/installed_scheduler_pool.rs

```diff
@@ -18,6 +38,59 @@ use {
 #[cfg(feature = "dev-context-only-utils")]
 use {mockall::automock, qualifier_attr::qualifiers};
 
+#[cfg_attr(feature = "dev-context-only-utils", automock)]
+pub trait InstalledSchedulerPool: Send + Sync + Debug {
+    fn take_from_pool(&self, context: SchedulingContext) -> Box<dyn InstalledScheduler>;
+    fn return_to_pool(&self, scheduler: Box<dyn InstalledScheduler>);
+}
+
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// Schedules, executes, and commits transactions under encapsulated implementation
+///
+/// The following chart illustrates the ownership/reference interaction between inter-dependent
+/// objects across crates:
+///
+/// ```mermaid
+/// graph TD
+///     Bank["Arc#lt;Bank#gt;"]
+///
+///     subgraph solana-runtime
+///         BankForks;
+///         BankWithScheduler;
+///         Bank;
+///         LoadExecuteAndCommitTransactions(["load_execute_and_commit_transactions()"]);
+///         SchedulingContext;
+///         InstalledSchedulerPool{{InstalledSchedulerPool}};
+///         InstalledScheduler{{InstalledScheduler}};
+///     end
+///
+///     subgraph solana-unified-scheduler-pool
+///         SchedulerPool;
+///         PooledScheduler;
+///         ScheduleExecution(["schedule_execution()"]);
+///     end
+///
+///     subgraph solana-ledger
+///         ExecuteBatch(["execute_batch()"]);
+///     end
+///
+///     ScheduleExecution -. calls .-> ExecuteBatch;
+///     BankWithScheduler -. dyn-calls .-> ScheduleExecution;
+///     ExecuteBatch -. calls .-> LoadExecuteAndCommitTransactions;
+///     linkStyle 0,1,2 stroke:gray,color:gray;
+///
+///     BankForks -- owns --> BankWithScheduler;
+///     BankForks -- owns --> InstalledSchedulerPool;
+///     BankWithScheduler -- refs --> Bank;
+///     BankWithScheduler -- owns --> InstalledScheduler;
+///     SchedulingContext -- refs --> Bank;
+///     InstalledScheduler -- owns --> SchedulingContext;
+///
+///     SchedulerPool -- owns --> PooledScheduler;
+///     SchedulerPool -. impls .-> InstalledSchedulerPool;
+///     PooledScheduler -. impls .-> InstalledScheduler;
+///     PooledScheduler -- refs --> SchedulerPool;
+/// ```
```

Comment:
fyi, this is rendered like this:

![image](https://github.com/solana-labs/solana/assets/117807/26367e1c-0dd2-43b2-a3bf-80bbbbe8baa4)